### PR TITLE
246 download and delete files

### PR DESCRIPTION
**Pull request by pinussilvestrus**

### CODE REVIEW

**Comment by pinussilvestrus** [comment]
[Download and Delete Files](https://trello.com/c/ZZpNWuII/246-download-and-delete-files)

File: src/modules/file-explorer/actions/file-explorer.js [hunk 1]
```diff
@@ -16,5 +33,30 @@ export default {
...
+			}).catch(err => {
+				Notification.showError(err.message);
+			});
+	},
+
+	delete: (file) => {
+		const currentUser = Server.get('user');
+		return s3Service.deleteFile(`users/${currentUser._id}`, file.name, null)
+			.then((res) => {
+				window.location.reload();
```
Comment (MasterCarl) [inline src/modules/file-explorer/actions/file-explorer.js]: TODO (See ticket)

File: src/modules/file-explorer/components/files.jsx [hunk 2]
```diff
@@ -8,27 +8,71 @@ class Files extends React.Component {
...
+			</div>
+		);
+	}
+
+	getFilesUI() {
+		let idCount = 0;
+		return (
+			<div>
+				{this.props.files.map((file) => {
+					idCount++;
```
Comment (MasterCarl) [inline src/modules/file-explorer/components/files.jsx]: I would assign every file its own ID

File: src/modules/file-explorer/containers/file-explorer.js [hunk 3]
```diff
@@ -10,13 +10,17 @@ const composer = (props, onData) => {
...
-		let componentData = {
-			actions,
-			files: res
-		};
-		onData(null, componentData);
-	}).catch(err => {
-		Notification.showError(err.message);
+			let componentData = {actions, files: []};
+
+			if( Object.prototype.toString.call( res ) === '[object Array]' ) {
```
Comment (MasterCarl) [inline src/modules/file-explorer/containers/file-explorer.js]: This looks like evil metaprogramming ;)

Comment (MasterCarl) [inline src/modules/file-explorer/containers/file-explorer.js]: If there's no other way to do it, I would at least add a comment

**Pull request merged**